functions/lib/pubsub/processUploads.js

```diff
@@ -152,3 +156,22 @@ function compressImage(tempFilePath, imageManipulation) {
     return imageManipulation.compress(tempFilePath, 960);
   }
 }
+
+function readImageData(mediaPath, localFS) {
+  return function() {
+    const imageData = {};
```

Comment:
Nit, this can be moved into the `then` function since it no longer needs to be shared between different scopes.